### PR TITLE
Rervert #488

### PR DESCRIPTION
**Pull request by samuelgarcia**
@alejoe91 @khl02007 
This PR revert #488 and make a better fix for the problem.
@khl02007 : could you test this ?

### CODE REVIEW

File: spikeinterface/core/segmentutils.py [hunk 1]
```diff
@@ -137,19 +137,22 @@ def __init__(self, recording_list, ignore_times=True, sampling_frequency_max_dif
                     assert d['t_start'] is None, ("ConcatenateSegmentRecording does not handle t_start. "
                                                   "Use ignore_times=True to ignore time information.")
                 parent_segments.append(parent_segment)
-        rec_seg = ProxyConcatenateRecordingSegment(parent_segments, ignore_times=ignore_times)
+        rec_seg = ProxyConcatenateRecordingSegment(parent_segments, one_rec.get_sampling_frequency(), ignore_times=ignore_times)
```
Comment (alejoe91) [inline spikeinterface/core/segmentutils.py:140]: can you use keyword here?

Comment (samuelgarcia) [reply spikeinterface/core/segmentutils.py:140]: sure ?

**Comment by khl02007** [comment]
@samuelgarcia @alejoe91 If the intention is to keep `sampling_frequency` and `times` mutually exclusive for the RecordingSegments, then I agree this is better. Next time I spot a problem, I'll just submit an issue rather than PR :)

Two questions:
* in that case,  why not do the same for Recordings? e.g. why do we calculate sampling rate in line 106 of nwbextractors even in the case where the timestamps have been specified? 
* somewhat related: in concatenated recording, why do we even have `ignore_times` parameter when it can only be `False`?

**Comment by samuelgarcia** [comment]
Hi Kyu.
No worry. This is quite new and not totally documented in the code.

The short story is: 
  * we have a `sampling_frequency` at `Recording` object level which is used for instance to design filter and compute the 
    duration. This sampling rate can be approximative.
  * we have at `RecordingSegment`:  `time_vector` OR `sampling_frequency` (optional `t_start`) only to reconstruct the times 
     vector. `times_kwargs` is for that case.

It sounds like a redundancy but internally it is very easier to handle sampling and times in 2 differents place.

NWB handle the 2 stuf in the init. So it increase the confusion.

`ignore_times` must be False only if there are some times information already (no often the case), so it force the user to be aware of this time reset. When no time_vector or no t_start ignore_times can stay False.

Does this PR fix your case ?

**Comment by khl02007** [comment]
> Does this PR fix your case ?

Yes it solves the problem I was having - thank you.

>we have a sampling_frequency at Recording object level which is used for instance to design filter and compute the
duration. This sampling rate can be approximative

why not estimate sampling rate only during filtering in that case? if the sampling rate is not the 'true' sampling rate, then it's a bit misleading to have it as a property of the recording. Is there any other place where an estimated sampling rate is handy?

> When no time_vector or no t_start ignore_times can stay False

If there is no time_vector or t_start, that means only sampling rate exists. in that case, what does `ignore_times` even mean? you don't have time information to ignore at all.

**Comment by samuelgarcia** [comment]
@khl02007 :
1. cool
2. because we need sampling rate at many place in fact not only filtering.
3. yes in that case there is no times no nothing to ignore. This option is only when threre is times.
    The `ignore_times` have to be read as `if_my_recording_have_time_vector_I_want_to_rest_the_time_on_purpose`
    The False by default is a good thing in my mind because it force user to think about it.

@alejoe91 : lets merge this ?

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**